### PR TITLE
add a dialog to choose active or standby position for radio frequency

### PR DESCRIPTION
**Pull request by AlphaLima**
image

image

image

image

image

image

image

### CODE REVIEW

**Review by brunotl — CHANGES_REQUESTED** [review]
android target don't build

File: Common/Source/Calc/Radio.cpp [hunk 1]
```diff
@@ -203,3 +203,21 @@ int SearchBestStation()
 	}
 	return Idx;
 }
+
+
+TCHAR* GetActiveStationSymbol(bool utf8_symbol) 
+{
+	if(utf8_symbol)
+		return((TCHAR*)ACTIVE_SYMBOL_UTF8) ;
```
Comment (brunotl) [inline Common/Source/Calc/Radio.cpp:211]: invalid cast

File: Common/Source/Calc/Radio.cpp [hunk 2]
```diff
@@ -203,3 +203,21 @@ int SearchBestStation()
...
+	else
+		return((TCHAR*)ACTIVE_SYMBOL);
```
Comment (brunotl) [inline Common/Source/Calc/Radio.cpp:213]: invalid cast

File: Common/Source/Calc/Radio.cpp [hunk 3]
```diff
@@ -203,3 +203,21 @@ int SearchBestStation()
...
+}
+
+TCHAR* GetStandyStationSymbol(bool utf8_symbol) 
+{
+	if(utf8_symbol)
+		return((TCHAR*)STANDBY_SYMBOL_UTF8) ;
```
Comment (brunotl) [inline Common/Source/Calc/Radio.cpp:219]: invalid cast

File: Common/Source/Calc/Radio.cpp [hunk 4]
```diff
@@ -203,3 +203,21 @@ int SearchBestStation()
...
+	else
+		return((TCHAR*)STANDBY_SYMBOL);
```
Comment (brunotl) [inline Common/Source/Calc/Radio.cpp:221]: invalid cast

File: Common/Source/Dialogs/dlgAirspaceDetails.cpp [hunk 5]
```diff
@@ -265,36 +288,54 @@ static void SetValues(WndForm* wf) {
...
+      }
+
       if(ValidFrequency(fASFrequency)) {
 
         WindowControl* wClose = wf->FindByName(TEXT("cmdClose"));
         if(wClose) {
-          wClose->SetLeft(IBLSCALE(115));
-          wClose->SetWidth(IBLSCALE(120));
+          wClose->SetLeft(IBLSCALE(155));
+          wClose->SetWidth(IBLSCALE(80));
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgAirspaceDetails.cpp:308]: why not in xml template ?

File: Common/Source/Dialogs/dlgAirspaceDetails.cpp [hunk 6]
```diff
@@ -265,36 +288,54 @@ static void SetValues(WndForm* wf) {
...
         }
 
-        wFreq->SetLeft(IBLSCALE(3));
-        wFreq->SetWidth(IBLSCALE(110));
+        wFreq->SetLeft(IBLSCALE(78));
+        wFreq->SetWidth(IBLSCALE(75));
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgAirspaceDetails.cpp:312]: why not in xml template ?

File: Common/Source/Dialogs/dlgAirspaceDetails.cpp [hunk 7]
```diff
@@ -265,36 +288,54 @@ static void SetValues(WndForm* wf) {
...
+
+        wSeqFreq->SetLeft(IBLSCALE(3));
+        wSeqFreq->SetWidth(IBLSCALE(75));
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgAirspaceDetails.cpp:315]: why not in xml template ?

File: Common/Source/Dialogs/dlgRadioPriSecSel.cpp [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+  LKASSERT( frmFreq !=NULL)
+  if(frmFreq) frmFreq->SetCaption(Name);
+
+
+  WndButton *wpnewActive = nullptr;
+  WndButton *wpnewPassive = nullptr; 
+  wpnewActive  = (WndButton*)wf->FindByName(TEXT("cmdSelActive"));
+  LKASSERT(   wpnewActive   !=NULL)
+  if(wpnewActive) 
+    wpnewActive->SetCaption((TCHAR*)GetActiveStationSymbol(Appearance.UTF8Pictorials));
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgRadioPriSecSel.cpp:109]: useless cast to TCHAR*

File: Common/Source/resource.h [hunk 9]
```diff
@@ -111,3 +111,4 @@
 #define IDR_XML_DEV_LXNAV_L         701
 #define IDR_XML_WEATHERSTDETAILS    702
 #define IDR_XML_NMEAREPLAY          703
+#define IDR_XML_RADIOPRISECSEL      704
```
Comment (brunotl) [inline Common/Source/resource.h:114]: missing end of file CR

File: Common/Source/Calc/Radio.cpp (hunk 1 above)
Comment (AlphaLima) [reply Common/Source/Calc/Radio.cpp]: fixed!

File: Common/Source/Calc/Radio.cpp (hunk 2 above)
Comment (AlphaLima) [reply Common/Source/Calc/Radio.cpp]: fixed!

File: Common/Source/Calc/Radio.cpp (hunk 3 above)
Comment (AlphaLima) [reply Common/Source/Calc/Radio.cpp]: fixed!

File: Common/Source/Dialogs/dlgAirspaceDetails.cpp (hunk 5 above)
Comment (AlphaLima) [reply Common/Source/Dialogs/dlgAirspaceDetails.cpp:305]: fixed!

File: Common/Source/Dialogs/dlgAirspaceDetails.cpp (hunk 6 above)
Comment (AlphaLima) [reply Common/Source/Dialogs/dlgAirspaceDetails.cpp]: fixed!

File: Common/Source/Dialogs/dlgRadioPriSecSel.cpp (hunk 8 above)
Comment (AlphaLima) [reply Common/Source/Dialogs/dlgRadioPriSecSel.cpp]: fixed!

File: Common/Source/resource.h (hunk 9 above)
Comment (AlphaLima) [reply Common/Source/resource.h]: fixed!

File: Common/Source/Dialogs/dlgWayPointDetails.cpp [hunk 10]
```diff
@@ -139,12 +139,34 @@ static void OnDetailsListInfo(WindowControl * Sender, WndListFrame::ListInfo_t *
...
+      const int subend = Surface.GetTextWidth(sTmp);
+
+      // size of the text is start of underline
+      sTmp[pos] = 0;
+      const int substart =  Surface.GetTextWidth(sTmp);
+
+      if(substart < subend) 
+      {
+        int h =  Surface.GetTextHeight(sTmp) - IBLSCALE(1);
+        const auto hOldPen = Surface.SelectObject(LKPen_Black_N2);
```
[the remark this answers is not in the document]
Comment (AlphaLima) [reply Common/Source/Dialogs/dlgWayPointDetails.cpp]: done

File: Common/Source/Calc/Radio.cpp [hunk 11]
```diff
@@ -52,6 +52,8 @@ BOOL ValidFrequency(double Freq)
 
 double  ExtractFrequencyPos(const TCHAR *text, size_t *start, size_t *len)
 {
+*start =0;
+*len =0;
```
[the remark this answers is not in the document]
Comment (AlphaLima) [reply Common/Source/Calc/Radio.cpp]: done and null pointer check

File: Common/Source/Dialogs/dlgAirspaceDetails.cpp (hunk 7 above)
Comment (AlphaLima) [reply Common/Source/Dialogs/dlgAirspaceDetails.cpp]: done

**Pull request merged**